test/cassandrax/keyspace_test.exs

```diff
@@ -0,0 +1,377 @@
+Code.require_file("../support/data_case.exs", __DIR__)
+
+defmodule TestData do
+  use Cassandrax.Schema
+
+  import Ecto.Changeset
+  import Cassandrax.Query
+
+  alias Cassandrax.TestConn
+
+  @type t :: %__MODULE__{}
+  @primary_key [:id]
+
+  table "test_data" do
+    field :id, :string
+    field :value, :string
+    field :svalue, MapSetType
+  end
+
+  def changeset(%__MODULE__{} = data, attrs \\ %{}) do
+    data
+    |> cast(attrs, [:id, :value])
+    |> validate_required([:id])
+  end
+
+  def create_table do
+    statement = [
+      "CREATE TABLE IF NOT EXISTS ",
+      "#{TestKeyspace.__keyspace__()}.test_data(",
+      "id text, ",
+      "value text, ",
+      "svalue set<text>, ",
+      "PRIMARY KEY (id))"
+    ]
+
+    {:ok, _result} = Cassandrax.cql(TestConn, statement)
+  end
+
+  def drop_table do
+    statement = "DROP TABLE IF EXISTS #{TestKeyspace.__keyspace__()}.test_data"
+
+    {:ok, _result} = Cassandrax.cql(TestConn, statement)
+  end
+end
+
+defmodule Cassandrax.KeyspaceTest do
+  use Cassandrax.DataCase
+
+  import Ecto.Changeset
+  import Cassandrax.Query
+
+  alias Cassandrax.TestConn
+  alias Cassandrax.Schema
+  alias Cassandrax.Keyspace.Batch
+  alias Ecto.Changeset
+
+  setup context do
+    TestData.create_table()
+    on_exit(fn -> TestData.drop_table() end)
+    seeds = Map.get(context, :seeds, [])
+    Enum.each(seeds, &TestKeyspace.insert!(&1))
+    :ok
+  end
+
+  defp schema_equal({:ok, a}, {:ok, b}), do: schema_equal(a, b)
+  defp schema_equal(a, b) do
+    Map.delete(a, :__meta__) == Map.delete(b, :__meta__)
+  end
+
+  defp list_sets_equal?(a, b) do
+    MapSet.equal?(MapSet.new(a), MapSet.new(b))
+  end
+
+  defp list_set_includes?(a, b) do
+    MapSet.subset?(MapSet.new(b), MapSet.new(a))
+  end
+
+  @zero %TestData{id: "0", value: "zero"}
+  @one %TestData{id: "1", value: "one", svalue: MapSet.new(["one", "another one"])}
+  @two %TestData{id: "2", value: "two"}
+  @three %TestData{id: "3", value: "three"}
+  @four %TestData{id: "4", value: "four"}
+  # invalid because 1 is an integer
+  @invalid_one %TestData{id: "1", value: 1}
+  @invalid_id %TestData{id: 1, value: 1}
+
+  describe "module functions" do
+    test "default option read: :one" do
+      assert TestKeyspace.__default_options__(:write) == [consistency: :one]
```

Comment:
you meant `TestKeyspace.__default_options__(:read)`

Reply:
yes i did